include/vcpkg/base/fmt.h

```diff
@@ -18,3 +18,11 @@ VCPKG_MSVC_WARNING(disable : 6239 4702)
 #include <fmt/format.h>
 #include <fmt/ranges.h>
 VCPKG_MSVC_WARNING(pop)
+
+template<class T>
+std::string adapt_to_string(const T& val)
```

Comment:
Should this be a constexpr functor object?

```cpp
constexpr struct AdaptToString
{
    template<class T>
    std::string operator()(const T& val) {
        std::string result;
        val.to_string(result);
        return result;
    }
} adapt_to_string;
```

Reply:
I don't believe anything else would want the name adapt_to_string to require such ADL defenses?

Reply:
Or, to put this another way: Maybe. To accomplish what?

Reply:
To make it "better" when used in functional style like `.map(adapt_to_string)`